src/biogeochem/CNDriverMod.F90

```diff
@@ -314,6 +317,14 @@ subroutine CNDriverNoLeaching(bounds,
     if (decomp_method == century_decomp) then
        call decomp_rate_constants_bgc(bounds, num_soilc, filter_soilc, &
             soilstate_inst, temperature_inst, ch4_inst, soilbiogeochem_carbonflux_inst)
+    else if (decomp_method == mimics_decomp) then
+       call decomp_rates_mimics(bounds, num_soilc, filter_soilc, &
+            soilstate_inst, temperature_inst, cnveg_carbonflux_inst, ch4_inst, &
+            soilbiogeochem_carbonflux_inst, soilbiogeochem_carbonstate_inst)
+    else
```

Comment:
This should already have been checked at initialization, so you don't really need to do this check here. And I'd suggest it should be an abort rather than a warning. Also no_soil_decomp is a valid option so it really shouldn't have an else...

Reply:
Removed `else`